include/TempoSyncKnob.h

```diff
@@ -44,6 +44,9 @@ class LMMS_EXPORT TempoSyncKnob : public Knob
 	TempoSyncKnob( KnobType knobNum, QWidget* parent = nullptr, const QString& name = QString() );
 	~TempoSyncKnob() override;
 
+	static TempoSyncKnob* buildLegacyKnob(KnobType knob_num, const QString& label, QWidget* parent);
+	static TempoSyncKnob* buildKnobWithSmallPixelFont(KnobType knob_num, const QString& label, QWidget* parent);
```

Comment:
Turning it into a constructor would lead to lots of verbose code in several areas. With this PR the `Knob` is intended to render the label using the font that's currently set for it. Without any intervention this normally would be the font that's used by its parent (as is also the case for Qt widgets).

Let's see what `Knob::buildKnobWithSmallPixelFont` does:
```
auto result = new Knob(knobNum, parent, name);
result->setFont(adjustedToPixelSize(result->font(), SMALL_FONT_SIZE));
result->setLabel(label);

return result;
```

One of the things that it does is to set the font to the small font size by using `adjustedToPixelSize`. This would have to be repeated in all places where clients currently use `buildKnobWithSmallPixelFont`. The label would also need to be set explicitly because the `Knob` constructor does not support taking the label. Therefore I propose to keep the builder/factory methods for now.

What do you think about moving the build methods that are currently provided by `Knob`, `CustomTextKnob` and `TempoSyncKnob` into a factory class?